File: remove_bg.py
from PIL import Image

def remove_white(image_path):
    img = Image.open(image_path).convert("RGBA")
    data = img.getdata()
    
    new_data = []
    for item in data:
        # Check if the pixel is near-white
        # R, G, B channels
        if item[0] > 240 and item[1] > 240 and item[2] > 240:
            new_data.append((255, 255, 255, 0)) # transparent
        else:
            new_data.append(item)
            
    img.putdata(new_data)
    img.save(image_path)

remove_white('src/assets/mascots/mascot_1_1.png')
print("Done")

File: scripts/remove_bg.mjs
import { Jimp } from "jimp";

async function main() {
  const image = await Jimp.read('src/assets/mascots/mascot_1_1.png');
  image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
    if(this.bitmap.data[idx + 0] > 240 && this.bitmap.data[idx + 1] > 240 && this.bitmap.data[idx + 2] > 240) {
      this.bitmap.data[idx + 3] = 0;
    }
  });
  await image.write('src/assets/mascots/mascot_1_1.png');
  console.log("Done");
}
main();
